hazelcast/src/main/java/com/hazelcast/transaction/impl/TransactionLog.java

```diff
@@ -116,7 +102,7 @@ public int size() {
 
     public List<Future> rollback(NodeEngine nodeEngine) {
         List<Future> futures = new ArrayList<Future>(size());
-        ListIterator<TransactionLogRecord> iterator = recordList.listIterator(size());
+        ListIterator<TransactionLogRecord> iterator = getRecordList().listIterator(size());
```

Comment:
I wonder if it's actually needs to be iterated in the reverse order. The `rollbackAsync()` iterates in forward order, so if this one can also be iterated in forward order, then we can avoid transforming `recordMap.values()` to an `ArrayList`, and just use `recordMap.values()` directly.

Reply:
I'm not sure about how it should be iterated but for now you can use `recordMap.values()` and keep the same behaviour. We can see about the iteration order later.